### DISCUSSION TITLE
red ring

### DISCUSSION THREAD

**Discussion by stubar** [leolabs/ableton-js#114] [Q&A]
Hi thanks for making a brilliant library!!

It it possible for this script to display a control surface red ring like you get in such scripts as the Push script?

**Comment by chozilla** [comment]
You mean the Area on the screen for the different control devices? 
Would be interesting if you could define this with javascript and you could highlight what is currently selected via script and modified there.

But then again with the script you can be everywhere at once. ;)

**Reply by stubar** [reply]
> You mean the Area on the screen for the different control devices?
Yes exactly, like you get with an APC40 for example.

Yes a lot of cases it wouldn't make sense but for me it does, I'm exposing a grid of cells and their track mixers to a web sockets client.

**Comment by jbflow** [comment]
@stubar 

This should be possible - [This resource](https://livecontrol.q3f.org/ableton-liveapi/articles/introduction-to-the-framework-classes/) is very old (it's the first one I came across when doing stuff with the remote scripts). But it talks about how to set up the red box session component using the framework classes. These have likely since been updated and there will be new classes contained within the v2 or even v3 inside the ableton folder in contained remote scripts.

**Comment by kkerti** [comment]
This is also on my radar. I think I'll work around this and just attach the required listeners in a grid pattern without actually showing the red ring. But after my proto is done, I will give this a go to include in abletonjs if no one makes it.

**Comment by kkerti** [comment]
I have started to work on this and got around a working prototype, following the code patterns of ableton-js. I'll open a pull request in November.
